infrastructure/tools/file/write.py

```
@@ -8,25 +9,46 @@
+                if filepath.startswith(prefix):
+                    filepath = filepath[len(prefix):]
+                    break
+            logger.info(f"Converted absolute path to relative: {filepath}")
         
+        target = (SANDBOX_DIR / filepath).resolve()
+
+        # 使用相对路径检查来验证沙箱边界
+        try:
+            target.relative_to(SANDBOX_DIR)
+        except ValueError:
+            raise ValueError(f"Access denied: Path outside sandbox. Requested: {filepath}, Resolved: {target}")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)